### PR TITLE
Markup generator \wo ugly double-switch

### PR DESCRIPTION
**Pull request by Piterden**
We got rid of switches, they were bad

### CODE REVIEW

**Comment by Piterden** [comment]
@wojpawlik What about the new variant?

File: markup.js [hunk 1]
```diff
@@ -168,43 +168,66 @@ class Markup {
...
+      },
+      {
+        condition: { type: 'text_link' },
+        startText: `<a href="${entity.url}">`,
+        endText: '</a>'
+      }
+    ].find((item) => Object.entries(item.condition)
+      .every(([key, value]) => typeof value === 'boolean'
+        ? Boolean(entity[key]) === value
+        : entity[key] === value))
```
Comment (wojpawlik) [inline markup.js:220]:
```suggestion
      .every(([key, value]) => entity[key] === value))
```

File: markup.js [hunk 2]
```diff
@@ -168,43 +168,66 @@ class Markup {
...
+        startText: '<code>',
+        endText: '</code>'
+      },
+      {
+        condition: { type: 'pre', language: true },
+        startText: `<pre><code class="language-${entity.language}">`,
+        endText: '</code></pre>'
+      },
+      {
+        condition: { type: 'pre', language: false },
```
Comment (wojpawlik) [inline markup.js:193]:
```suggestion
        condition: { type: 'pre', language: undefined },
```

File: markup.js [hunk 3]
```diff
@@ -168,43 +168,66 @@ class Markup {
...
+        startText: '<i>',
+        endText: '</i>'
+      },
+      {
+        condition: { type: 'code' },
+        startText: '<code>',
+        endText: '</code>'
+      },
+      {
+        condition: { type: 'pre', language: true },
```
Comment (wojpawlik) [inline markup.js:188]:
```suggestion
        condition: { type: 'pre' },
```

**Comment by Piterden** [comment]
Damn, Travis is broken @wojpawlik

**Comment by wojpawlik** [comment]
Yeah, need to reorder the 2 cases

**Pull request closed without merging**